sdk/include/opentelemetry/sdk/metrics/aggregation/default_aggregation.h

```diff
@@ -159,7 +160,29 @@ class DefaultAggregation
               new DoubleSumAggregation(nostd::get<SumPointData>(point_data)));
         }
       default:
-        return DefaultAggregation::CreateAggregation(instrument_descriptor, nullptr);
+        return nullptr;  // won't reach here
+    }
+  }
+
+  static AggregationType GetDefaultAggregationType(InstrumentType instrument_type,
+                                                   bool &is_monotonic)
+  {
+    is_monotonic = true;
+    switch (instrument_type)
+    {
+      case InstrumentType::kCounter:
+      case InstrumentType::kObservableCounter:
+        is_monotonic = false;
+        return AggregationType::kSum;
+      case InstrumentType::kUpDownCounter:
+      case InstrumentType::kObservableUpDownCounter:
+        return AggregationType::kSum;
+      case InstrumentType::kHistogram:
+        return AggregationType::kHistogram;
+      case InstrumentType::kObservableGauge:
+        return AggregationType::kLastValue;
+      default:
+        return AggregationType::kDrop;
     }
   }
 };
```

Comment:
I don't understand the semantic here, what does `is_monotonic` means ?

Counter and Observable counter are not monotonic but everything else is ?

This looks like inverted logic (compared to my understanding of a counter).

Please clarify.

Reply:
Your understanding is correct :) The logic is incorrectly inverted in this PR. And since we don't have negative values in UpDownCounter unit-test, there was no error. I have now changed the unit-test to record negative values, and let the test fail first before fixing.

Reply:
Ok, I can see the test failing now:
```
The following tests FAILED:
Errors while running CTest
	313 - metrics.*/WritableMetricStorageTestUpDownFixture.TestAggregation/* (Failed)
```

Adding fix now.

Reply:
Added fix. The unit-test is passing now.
Thanks for noticing the issue.